File: src/main/resources/templates/swift/solution/Sources/swiftExercise/Policy.swift
import Foundation

public class Policy {
	private var context: Context!

	init(_ context: Context!) {
		self.context = context
	}

	public func configure() {
		if self.context.getDates().count > 10 {
			print("More than 10 dates, choosing merge sort!")
			self.context.setSortAlgorithm(MergeSort())
		} else {
			print("Less or equal than 10 dates. choosing quick sort!")
			self.context.setSortAlgorithm(BubbleSort())
		}
	}
}

File: src/main/resources/templates/java/maven_blackbox/solution/src/${packageNameFolder}/model/Policy.java
package ${packageName}.model;

public class Policy {

    /**
     * @oracleIgnore
     */
    private static final int VALUES_SIZE_THRESHOLD = 10;

    private final Context context;

    public Policy(Context context) {
        this.context = context;
    }

    /**
     * Chooses a strategy depending on the number of Dates.
     */
    public void configure() {
        if (this.context.getDates().size() > VALUES_SIZE_THRESHOLD) {
            System.out.println("More than " + VALUES_SIZE_THRESHOLD
                    + " double values, choosing merge sort!");
            this.context.setSortAlgorithm(new MergeSort());
        } else {
            System.out.println("Less or equal than " + VALUES_SIZE_THRESHOLD
                    + " double values, choosing quick sort!");
            this.context.setSortAlgorithm(new BubbleSort());
        }
    }
}
